File: backend/app/schemas/triage.py
from typing import Literal

from pydantic import BaseModel, Field


Severity = Literal["Critical", "Moderate", "Low"]


class TriageAnalyzeRequest(BaseModel):
    symptoms: str = Field(min_length=3, max_length=2000)
    age: int = Field(ge=0, le=120)
    gender: str = Field(min_length=1, max_length=50)
    existing_conditions: str = Field(default="", max_length=1000)


class TriageAnalyzeResponse(BaseModel):
    severity: Severity
    confidence: float = Field(ge=0, le=1)
    recommended_action: str
    recommended_hospital_type: str
    ambulance_required: bool
    matched_keywords: list[str]

File: frontend/lib/triage.ts
import { apiClient } from "@/lib/api-client";

export type TriageAnalyzeRequest = {
  symptoms: string;
  age: number;
  gender: string;
  existing_conditions: string;
};

export type TriageAnalyzeResponse = {
  severity: "Critical" | "Moderate" | "Low";
  confidence: number;
  recommended_action: string;
  recommended_hospital_type: string;
  ambulance_required: boolean;
  matched_keywords: string[];
};

export async function analyzeTriage(
  payload: TriageAnalyzeRequest,
): Promise<TriageAnalyzeResponse> {
  return apiClient<TriageAnalyzeResponse>("/api/triage/analyze", {
    errorMessage: "Unable to analyze symptoms",
    method: "POST",
    body: payload,
  });
}
